File: python/examples/list_laws.py
"""List all laws for a country, auto-paginated.

Usage::

    LEGALIZE_API_KEY=leg_... python examples/list_laws.py es ley_organica
"""

from __future__ import annotations

import os
import sys

from legalize import Legalize


def main() -> int:
    country = sys.argv[1] if len(sys.argv) > 1 else "es"
    law_type = sys.argv[2] if len(sys.argv) > 2 else None

    with Legalize(api_key=os.environ["LEGALIZE_API_KEY"]) as client:
        for i, law in enumerate(client.laws.iter(country=country, law_type=law_type), start=1):
            print(f"{i:5}  {law.id}  {law.title[:80]}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: node/examples/list_laws.ts
/**
 * List all laws for a country, auto-paginated.
 *
 * Usage:
 *
 *   LEGALIZE_API_KEY=leg_... npx tsx examples/list_laws.ts es ley_organica
 */

import { Legalize } from "@legalize-dev/sdk";

async function main(): Promise<number> {
  const country = process.argv[2] ?? "es";
  const lawType = process.argv[3];

  const client = new Legalize();
  let i = 1;
  try {
    for await (const law of client.laws.iter(country, lawType ? { lawType } : {})) {
      console.log(`${String(i).padStart(5)}  ${law.id}  ${(law.title ?? "").slice(0, 80)}`);
      i += 1;
    }
  } finally {
    await client.close();
  }
  return 0;
}

main().then((code) => process.exit(code));
